scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -95,11 +97,54 @@ namespace <%= namespace %> {
         }
 
         public void CheckoutWithNativeWebView(CheckoutSuccessCallback success, CheckoutFailureCallback failure, CheckoutCancelCallback cancelled) {
-            GetWebCheckoutLink(url => {
-                WebCheckout.Checkout(url, success, failure, cancelled);
-            }, error => {
-                failure(error);
-            });
+            if (WebCheckout != null) {
+                GetWebCheckoutLink(url => {
+                    WebCheckout.Checkout(url, success, failure, cancelled);
+                }, error => {
+                    failure(error);
+                });
+            } else {
+                // TODO throw an exception,  this platform does not support checkout with web pay
```

Comment:
Ideally we should have a `CanCheckoutWithNativeWeb` because this feature isn't available on Android. A Failure callback means a Web checkout started but failed for some temporary reason `HTTP Error` etc. An exception in this case means you can't checkout at all, and it was the programmer's fault for misleading the user to do so